docs: add supported connection paths for remote clusters (CCS/CCR vs private connectivity) (#7567)
elastic/docs-content: deploy-manage/remote-clusters.md

| **Elastic Cloud Enterprise** | [IP filter](/deploy-manage/security/ip-filtering.md) | [Remote cluster filter](/deploy-manage/security/remote-cluster-filtering.md) / [IP filter](/deploy-manage/security/ip-filtering.md) (\*) | [IP filter](/deploy-manage/security/ip-filtering.md) or [Kubernetes network policy](/deploy-manage/security/k8s-network-policies.md) |
| **Self-managed / Elastic Cloud on Kubernetes** | [IP filter](/deploy-manage/security/ip-filtering.md) | [IP filter](/deploy-manage/security/ip-filtering.md) | [IP filter](/deploy-manage/security/ip-filtering.md) or [Kubernetes network policy](/deploy-manage/security/k8s-network-policies.md) |

(*) For ECE, remote cluster filters apply when both clusters are in the **same environment**. Use IP filters when the clusters belong to **different environments**.

::::{note}
When using self-managed security mechanisms (such as firewalls), keep in mind that remote clusters with API key–based authentication use port `9443` by default. Specify this port if a destination port is required.
::::

> ### Connection paths and private connectivity [remote-clusters-connection-paths]

Reviewer: Is the CRLF / ^M an issue or will the doc system work with both? 
Just noticed that GH diff highlights this in red ...

Reviewer 2: it breaks things (a table further down is broken). I'm removing now

Author: Fixed — the CRLF noise came from my original commit normalizing the file's mixed line endings. I've rebuilt the branch so the file keeps main's original bytes and the new section is appended with LF endings: the diff is now a clean ~22-line addition with no ^M highlights.

> ### Connection paths and private connectivity [remote-clusters-connection-paths]

Reviewer 2: ```suggestion
### Connection paths and private connectivity [remote-clusters-connection-paths]
```

Remote cluster connections to an ECH deployment can use either the remote deployment’s public proxy address or a [private endpoint](/deploy-manage/security/private-connectivity.md), depending on where the local cluster runs. When the local cluster is an ECH deployment, the connection must use public endpoints. Private connectivity is supported only when the local cluster runs in your own VPC or VNet, such as a self-managed, {{eck}}, or {{ece}} cluster.

> The following table summarizes the supported connection paths:

Reviewer 2: ```suggestion
Remote cluster connections to an ECH deployment can use either the remote deployment’s public proxy address or a [private endpoint](/deploy-manage/security/private-connectivity.md), depending on where the local cluster runs. When the local cluster is an ECH deployment, the connection must use public endpoints. Private connectivity is supported only when the local cluster runs in your own VPC or VNet, such as a self-managed, {{eck}}, or {{ece}} cluster.

The following table summarizes the supported connection paths:

```

| Local cluster location | Remote cluster | Connection path | Private connectivity applicable? |
|---|---|---|---|
| {{ech}} deployment (any region or cloud provider) | {{ech}} deployment | Remote proxy address over public endpoints. Cross-region and cross-provider connections are supported. | **No.** Private connection hostnames resolve only inside your VPC or VNet through your private DNS zone. Deployment-to-deployment traffic originates from Elastic-managed networks and can't traverse your private endpoint. |

> | Self-managed or {{ece}} cluster in your VPC or VNet | {{ech}} deployment | Public proxy address, or a private connection: create a VPC endpoint (AWS), private endpoint (Azure), or Private Service Connect endpoint (GCP) and connect to the remote cluster through it. Inter-region private connections are supported where the cloud provider supports them. | **Yes** (in this direction only). |

Reviewer 2: ```suggestion
| Self-managed or {{ece}} cluster in your VPC or VNet | {{ech}} deployment | Public proxy address, or a private connection: create a VPC endpoint (AWS), private endpoint (Azure), or Private Service Connect endpoint (GCP) and connect to the remote cluster through it. Inter-region private connections are supported where the cloud provider supports them. | **Yes** (in this direction only). |
```

| {{ech}} deployment | Self-managed cluster in your network | Outbound from Elastic over the public internet to your cluster's endpoint. Allow the connection with IP-based rules on your side. | **No.** {{ech}} deployments can't consume private endpoints in customer networks. |

::::{warning}

> Don't use a private connection hostname (for example, `*.vpce.{region}.aws.elastic-cloud.com` or your Azure private hosted zone domain) as the `proxy_address` for a remote cluster connection between two {{ech}} deployments. The connection will fail because the hostname is not resolvable from Elastic-managed networks. Use the public proxy address from the remote deployment's **Security** page instead.

Reviewer: Is there a specific error for this? Something like "can't resolve" or "host unknown" or so? Would be great to add the symptom/error so users can more easily find the doc.

Author: Added. The symptom is a DNS resolution failure — `unknown host` / `UnknownHostException` in the Elasticsearch logs — since the private hostname isn't resolvable from Elastic-managed networks. The warning now includes it for searchability.

> Don't use a private connection hostname (for example, `*.vpce.{region}.aws.elastic-cloud.com` or your Azure private hosted zone domain) as the `proxy_address` for a remote cluster connection between two {{ech}} deployments. The connection will fail because the hostname is not resolvable from Elastic-managed networks. Use the public proxy address from the remote deployment's **Security** page instead.

Reviewer 2: ```suggestion
Don't use a private connection hostname (for example, `*.vpce.{region}.aws.elastic-cloud.com` or your Azure private hosted zone domain) as the `proxy_address` for a remote cluster connection between two {{ech}} deployments. The connection will fail because the hostname is not resolvable from Elastic-managed networks. Use the public proxy address from the remote deployment's **Security** page instead.
```

::::

::::{tip}

> Traffic sent between deployments for {{ccs}} and {{ccr}} is metered as **Data out** on the deployment the data is replicated or searched from, regardless of destination or path. Refer to [Cloud Hosted deployment billing dimensions](/deploy-manage/cloud-organization/billing/cloud-hosted-deployment-billing-dimensions.md).

Reviewer: > or searched from

I think this is ambiguous and might cause confusion. For CCS is this the cluster initiating the search or the one that executes it?

Author: Good catch — reworded. It's the remote deployment being searched (the search results leave that deployment), not the initiating cluster. The tip now spells out both directions: for CCR the deployment you replicate from, for CCS the remote deployment being searched.

(*) For ECE, remote cluster filters apply when both clusters are in the **same environment**. Use IP filters when the clusters belong to **different environments**.

::::{note}
When using self-managed security mechanisms (such as firewalls), keep in mind that remote clusters with API key–based authentication use port `9443` by default. Specify this port if a destination port is required.
::::

### Connection paths and private connectivity [remote-clusters-connection-paths]

Remote cluster connections to an ECH deployment can use either the remote deployment’s public proxy address or a [private endpoint](/deploy-manage/security/private-connectivity.md), depending on where the local cluster runs. When the local cluster is an ECH deployment, the connection must use public endpoints. Private connectivity is supported only when the local cluster runs in your own VPC or VNet, such as a self-managed, {{eck}}, or {{ece}} cluster.

> The following table summarizes the supported connection paths:

Reviewer 2: ```suggestion
The following table summarizes the supported connection paths:

```

(*) For ECE, remote cluster filters apply when both clusters are in the **same environment**. Use IP filters when the clusters belong to **different environments**.

::::{note}
When using self-managed security mechanisms (such as firewalls), keep in mind that remote clusters with API key–based authentication use port `9443` by default. Specify this port if a destination port is required.
::::

### Connection paths and private connectivity [remote-clusters-connection-paths]

Remote cluster connections to an ECH deployment can use either the remote deployment’s public proxy address or a [private endpoint](/deploy-manage/security/private-connectivity.md), depending on where the local cluster runs. When the local cluster is an ECH deployment, the connection must use public endpoints. Private connectivity is supported only when the local cluster runs in your own VPC or VNet, such as a self-managed, {{eck}}, or {{ece}} cluster.

The following table summarizes the supported connection paths:

The following table summarizes the supported connection paths:

> | Self-managed or {{ece}} cluster in your VPC or VNet | {{ech}} deployment | Public proxy address, or a private connection: create a VPC endpoint (AWS), private endpoint (Azure), or Private Service Connect endpoint (GCP) and connect to the remote cluster through it. Inter-region private connections are supported where the cloud provider supports them. | **Yes** (in this direction only). |

Reviewer 2: ```suggestion

```

| Local cluster location | Remote cluster | Connection path | Private connectivity applicable? |
|---|---|---|---|
| {{ech}} deployment (any region or cloud provider) | {{ech}} deployment | Remote proxy address over public endpoints. Cross-region and cross-provider connections are supported. | **No.** Private connection hostnames resolve only inside your VPC or VNet through your private DNS zone. Deployment-to-deployment traffic originates from Elastic-managed networks and can't traverse your private endpoint. |
Don't use a private connection hostname (for example, `*.vpce.{region}.aws.elastic-cloud.com` or your Azure private hosted zone domain) as the `proxy_address` for a remote cluster connection between two {{ech}} deployments. The connection will fail because the hostname is not resolvable from Elastic-managed networks. Use the public proxy address from the remote deployment's **Security** page instead.

> 

Reviewer 2: ```suggestion
| Self-managed or {{ece}} cluster in your VPC or VNet | {{ech}} deployment | Public proxy address, or a private connection: create a VPC endpoint (AWS), private endpoint (Azure), or Private Service Connect endpoint (GCP) and connect to the remote cluster through it. Inter-region private connections are supported where the cloud provider supports them. | **Yes** (in this direction only). |
```

::::{note}
When using self-managed security mechanisms (such as firewalls), keep in mind that remote clusters with API key–based authentication use port `9443` by default. Specify this port if a destination port is required.
::::

> ### Supported connection paths [remote-clusters-connection-paths]

Reviewer 3: ```suggestion
### Connection paths and private connectivity [remote-clusters-connection-paths]
```

> Remote cluster connections to an {{ech}} deployment always target the remote deployment's **proxy address**, found on the **Security** page of the remote deployment: port `9400` for the TLS certificate–based model, or port `9443` for the API key–based model. The network paths supported for this connection depend on where the local cluster runs:

Reviewer 3: > Remote cluster connections to an {{ech}} deployment always target the remote deployment's **proxy address**

That's contradictory with the messages that we add in the limitation snippet and in the table below.
If the local cluster is within the same VPC as the private-link, the user could potentially use the private endpoint, right? At least that's explained in other parts of this PR if I'm not mistaken.

Let me know what you think about this suggestion:

```suggestion
Remote cluster connections to an ECH deployment can use either the remote deployment’s public proxy address or a [private endpoint](/deploy-manage/security/private-connectivity.md), depending on where the local cluster runs. When the local cluster is an ECH deployment, the connection must use public endpoints. Private connectivity is supported only when the local cluster runs in your own VPC or VNet, such as a self-managed, {{eck}}, or {{ece}} cluster.

The following table summarizes the supported connection paths:
```

Reviewer 3: Note that I have removed from this statement authentication methods (API key or TLS certs) and ports (9443 and 9400), as they should be out of the scope of this section, and they are explained in many other documents.

If we consider it's very useful we can add an extra paragraph. I know a lot of users and customers are always confused between 9400 and 9443, but we have clarified that in a lot of places already.

| Local cluster location | Remote cluster | Connection path | Private connectivity applicable? |
|---|---|---|---|
| {{ech}} deployment (any region or cloud provider) | {{ech}} deployment | Remote proxy address over public endpoints. Cross-region and cross-provider connections are supported. | **No.** Private connection hostnames resolve only inside your VPC or VNet through your private DNS zone. Deployment-to-deployment traffic originates from Elastic-managed networks and can't traverse your private endpoint. |

> | Self-managed or {{ece}} cluster in your VPC or VNet | {{ech}} deployment | Public proxy address, or a private connection: create a VPC endpoint (AWS), private endpoint (Azure), or Private Service Connect endpoint (GCP) and connect to the remote cluster ports through it. Inter-region private connections are supported where the cloud provider supports them. | **Yes** (in this direction only). |

Reviewer 3: ```suggestion
| Self-managed or {{ece}} cluster in your VPC or VNet | {{ech}} deployment | Public proxy address, or a private connection: create a VPC endpoint (AWS), private endpoint (Azure), or Private Service Connect endpoint (GCP) and connect to the remote cluster through it. Inter-region private connections are supported where the cloud provider supports them. | **Yes** (in this direction only). |
```

| {{ech}} deployment | Self-managed cluster in your network | Outbound from Elastic over the public internet to your cluster's endpoint. Allow the connection with IP-based rules on your side. | **No.** {{ech}} deployments can't consume private endpoints in customer networks. |

::::{warning}

> Don't use a private connection hostname (for example, `*.vpce.{region}.aws.elastic-cloud.com` or your Azure private hosted zone domain) as the `proxy_address` for a remote cluster connection between two {{ech}} deployments. The connection will fail because the hostname is not resolvable from Elastic-managed networks. Use the proxy address from the remote deployment's **Security** page instead.

Reviewer 3: ```suggestion
Don't use a private connection hostname (for example, `*.vpce.{region}.aws.elastic-cloud.com` or your Azure private hosted zone domain) as the `proxy_address` for a remote cluster connection between two {{ech}} deployments. The connection will fail because the hostname is not resolvable from Elastic-managed networks. Use the public proxy address from the remote deployment's **Security** page instead.
```

::::{note}
When using self-managed security mechanisms (such as firewalls), keep in mind that remote clusters with API key–based authentication use port `9443` by default. Specify this port if a destination port is required.
::::

### Connection paths and private connectivity [remote-clusters-connection-paths]

Remote cluster connections to an ECH deployment can use either the remote deployment’s public proxy address or a [private endpoint](/deploy-manage/security/private-connectivity.md), depending on where the local cluster runs. When the local cluster is an ECH deployment, the connection must use public endpoints. Private connectivity is supported only when the local cluster runs in your own VPC or VNet, such as a self-managed, {{eck}}, or {{ece}} cluster.

The following table summarizes the supported connection paths:

| Local cluster location | Remote cluster | Connection path | Private connectivity applicable? |
|---|---|---|---|

> | {{ech}} deployment (any region or cloud provider) | {{ech}} deployment | Remote proxy address over public endpoints. Cross-region and cross-provider connections are supported. | **No.** Private connection hostnames resolve only inside your VPC or VNet through your private DNS zone. Deployment-to-deployment traffic originates from Elastic-managed networks and can't traverse your private endpoint. |

Reviewer 4: The problem is not that the hostnames would not resolve. Ther problem is that the user does not have permission to create them inside the Elastic CLoud.
```suggestion
| {{ech}} deployment (any region or cloud provider) | {{ech}} deployment | Remote proxy address over public endpoints. Cross-region and cross-provider connections are supported. | **No.** You can only create private connection hostnames inside your VPC or VNet through your private DNS zone. Deployment-to-deployment traffic originates from Elastic-managed networks and can't traverse your private endpoint. |
```

> | Self-managed or {{ece}} cluster in your VPC or VNet | {{ech}} deployment | Public proxy address, or a private connection: create a VPC endpoint (AWS), private endpoint (Azure), or Private Service Connect endpoint (GCP) and connect to the remote cluster through it. Inter-region private connections are supported where the cloud provider supports them. | **Yes** (in this direction only). |

Reviewer 4: To keep capitalization common across services
```suggestion
| Self-managed or {{ece}} cluster in your VPC or VNet | {{ech}} deployment | Public proxy address, or a private connection: create a VPC Endpoint (AWS), Private Endpoint (Azure), or Private Service Connect endpoint (GCP) and connect to the remote cluster through it. Inter-region private connections are supported where the cloud provider supports them. | **Yes** (in this direction only). |
```

> | Self-managed or {{ece}} cluster in your VPC or VNet | {{ech}} deployment | Public proxy address, or a private connection: create a VPC endpoint (AWS), private endpoint (Azure), or Private Service Connect endpoint (GCP) and connect to the remote cluster through it. Inter-region private connections are supported where the cloud provider supports them. | **Yes** (in this direction only). |

Reviewer 5: I don't think that "in this direction only" is very clear. What direction?

```suggestion
| Self-managed or {{ece}} cluster in your VPC or VNet | {{ech}} deployment | Public proxy address, or a private connection: create a VPC endpoint (AWS), private endpoint (Azure), or Private Service Connect endpoint (GCP) and connect to the remote cluster through it. Inter-region private connections are supported where the cloud provider supports them. | **Yes** (customer network → ECH only). |
```

> | {{ech}} deployment | Self-managed cluster in your network | Outbound from Elastic over the public internet to your cluster's endpoint. Allow the connection with IP-based rules on your side. | **No.** {{ech}} deployments can't consume private endpoints in customer networks. |

Reviewer 4: ```suggestion
| {{ech}} deployment | Self-managed cluster in your network | Outbound from Elastic over the public internet to your cluster's endpoint. Allow the connection with IP-based rules on your side. | **No.** {{ech}} deployments can't create private endpoints for customer networks. |
```
